hawc/apps/assessment/admin.py

```diff
@@ -33,12 +34,30 @@ class AssessmentAdmin(admin.ModelAdmin):
         "reviewers__last_name",
     )
 
-    actions = (bust_cache,)
+    actions = (bust_cache, "delete_orphan_tags")
 
     def queryset(self, request):
         qs = super().queryset(request)
         return qs.prefetch_related("project_manager", "team_members", "reviewers")
 
+    def delete_orphan_tags(self, request, queryset):
+        # Action can only be run on one assessment at a time
+        if queryset.count() != 1:
+            self.message_user(
+                request, f"Select only one item to perform the action on.", level=messages.WARNING
+            )
+            return
+        assessment = queryset.first()
+        root = ReferenceFilterTag.get_assessment_root(assessment.id)
+        # queryset of tag tree associated with assessment
+        filter_tags = ReferenceFilterTag.get_tree(root)
+        # queryset of tags associated with assessment
+        tags = ReferenceTags.objects.filter(content_object__assessment_id=assessment.id)
+        # delete tags that are not in the assessment tag tree
+        deleted, _ = tags.exclude(tag__in=filter_tags).delete()
```

Comment:
we should probably log somewhere what was deleted in case it ever needs to be recovered. Perhaps we can write in the assessment.Log model an item which just lists hte reference-id and tag-id of what was deleted?